### ISSUE TITLE
How to use for graph level unsupervised clustering, not the node.

### ISSUE THREAD

**Issue by PeterDeSOM** [shaham-lab/SpectralNet#2]
I have millions of graph data that cannot be labeled by human efforts. I need it to be classified by unsupervised technique, not semi and not the node level. Is it possible?

**Comment by AmitaiYacobi** [comment]
Hi! Thanks for reaching out.
 
If you'd like to **cluster** the data or find any similarity between data points, you can definitely use our method for doing so.
What do you mean by "graph level"? that each data point (sample) is a graph?

**Comment by PeterDeSOM** [comment]
Thank you for your comment,

Graph-Level means that each data point (sample) is a graph, you are right. All that I've found the method for Graph Clustering is actually for node clustering or not a fully unsupervised learning method. 

For example, it is;
- pretraining some datasets with the unlabeled using the contrastive learning method or others. (No output channel or number of classes to be clustered is required)
- then finetuning the dataset with its labels using the supervised learning method.

It's not a fully unsupervised learning manner.

Each of my graph data has 80 ~ 100 nodes and their edges. As I told you before, I have that graph data over millions and I want it to be classified by unsupervised learning method that doesn't need its own label to train. I want to classify it automatically by the unsupervised learning method(model).

I couldn't find the right model yet or I may not know how to. That's why I contact you to find a solution.
Could you show me a little direction how to?

Thank you.

**Comment by AmitaiYacobi** [comment]
Hi!
As long as you can find a good embedding for a graph, i.e., a function that will embed the graph into a euclidean space for instance, you can feed the model with those embeddings and it should be working.

**Comment by PeterDeSOM** [comment]
Thank you for your comment,

Do you mean like the KMean algorithm can be used for it?

**Comment by AmitaiYacobi** [comment]
No. 
The input to the networks is basically an embedding of the original data so that each input data point (in your case a graph) is represented as a vector.

If you will find a model/function/method that can take a graph and represent it as a vector, you will be able to use these graph representations as input to SpectralNet.
